leader-election-impl/src/main/java/com/palantir/paxos/PaxosQuorumChecker.java

```diff
@@ -249,9 +243,11 @@ private static boolean timedOut(Future<?> responseFuture) {
     }
 
     private static <SERVICE, RESPONSE extends PaxosResponse> void cancelOutstandingRequestsAfterTimeout(
+            boolean shouldCancel,
             List<Future<Map.Entry<SERVICE, RESPONSE>>> responseFutures) {
+
         boolean areAllRequestsComplete = responseFutures.stream().allMatch(Future::isDone);
-        if (areAllRequestsComplete) {
+        if (!shouldCancel || areAllRequestsComplete) {
```

Comment:
nit: ok. I'd probably just not call this method if `shouldCancel` is false above, but this is fine.